block-producer/src/state_view/mod.rs

```diff
@@ -54,7 +56,22 @@ where
         &self,
         candidate_tx: &ProvenTransaction,
     ) -> Result<(), VerifyTxError> {
-        // 1. soft-check if `tx` violates in-flight requirements.
+        #[cfg(not(test))]
```

Comment:
I'm not sure which test is failing, but this probably means this class should be generic over the `TransactionVerifier` implementation. Then:

- The code bellow would be a `DefaultVerifier` of sorts, that runs the actual verify code.
- The `DummyProver` would have two verifier counter parts. Possible a `SuccessfulDummyVerifier` and a `FailureDummyVerifier`. Then the tests would use this implementation.

The changes above would allow one to test the code path were a transaction verification fails. And would save us the time to generate an actual proof in the tests.

Reply:
Yes, agree that we should handle it via traits. Let's do this in a different PR though.

Reply:
Here is the issue: https://github.com/0xPolygonMiden/miden-node/issues/229